openwisp_monitoring/check/tests/test_models.py

```diff
@@ -42,34 +56,128 @@ def test_base_check_class(self):
 
     def test_check_instance(self):
         obj = self._create_device(organization=self._create_org())
-        c = Check(
-            name='Ping class check', check=self._PING, content_object=obj, params={}
-        )
-        i = c.check_instance
-        self.assertIsInstance(i, Ping)
-        self.assertEqual(i.related_object, obj)
-        self.assertEqual(i.params, c.params)
+        with self.subTest('Test Ping check instance'):
+            c = Check(
+                name='Ping class check', check=self._PING, content_object=obj, params={}
+            )
+            i = c.check_instance
+            self.assertIsInstance(i, Ping)
+            self.assertEqual(i.related_object, obj)
+            self.assertEqual(i.params, c.params)
+        with self.subTest('Test Configuration Applied check instance'):
+            c = Check(
+                name='Configuration Applied class check',
+                check=self._CONFIG_APPLIED,
+                content_object=obj,
+                params={},
+            )
+            i = c.check_instance
+            self.assertIsInstance(i, ConfigApplied)
+            self.assertEqual(i.related_object, obj)
+            self.assertEqual(i.params, c.params)
 
     def test_validation(self):
-        check = Check(name='Ping check', check=self._PING, params={})
-        try:
-            check.full_clean()
-        except ValidationError as e:
-            self.assertIn('device', str(e))
-        else:
-            self.fail('ValidationError not raised')
-
-    def test_auto_ping(self):
+        with self.subTest('Test Ping check validation'):
+            check = Check(name='Ping check', check=self._PING, params={})
+            try:
+                check.full_clean()
+            except ValidationError as e:
+                self.assertIn('device', str(e))
+            else:
+                self.fail('ValidationError not raised')
+        with self.subTest('Test Configuration Applied check validation'):
+            check = Check(name='Config check', check=self._CONFIG_APPLIED, params={})
+            try:
+                check.full_clean()
+            except ValidationError as e:
+                self.assertIn('device', str(e))
+            else:
+                self.fail('ValidationError not raised')
+
+    def test_auto_check_creation(self):
         self.assertEqual(Check.objects.count(), 0)
         d = self._create_device(organization=self._create_org())
-        self.assertEqual(Check.objects.count(), 1)
-        c = Check.objects.first()
-        self.assertEqual(c.content_object, d)
-        self.assertIn('Ping', c.check)
+        self.assertEqual(Check.objects.count(), 2)
+        with self.subTest('Test AUTO_PING'):
+            c1 = Check.objects.get(name='Ping')
+            self.assertEqual(c1.content_object, d)
+            self.assertIn('Ping', c1.check)
+        with self.subTest('Test AUTO_CONFIG_CHECK'):
+            c2 = Check.objects.get(name='Configuration Applied')
```

Comment:
don't query by name, use `Check.objects.filter(check=self._CONFIG_APPLIED).first()`

fix the same issue for all other occurrences across the codebase!